subprojects/packagefiles/theora/meson.build

```diff
@@ -0,0 +1,88 @@
+project(
+  'theora',
+  'c', 'cpp',
+  version: '1.1.1',
+  meson_version: '>= 0.59.0',
+  license: 'BSD-3-Clause',
+  default_options: ['c_std=c11']
+)
+
+conf = configuration_data()
+
+compile_args = []
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() != 'msvc'
```

Comment:
I've had arguments about this before. I think the core developers are in favor of this approach.

OTOH, get_supported_arguments checks that the arguments can be actually used, which implies it's safe to use without cc.get_argument_syntax() != 'msvc'

Reply:
Those checks are an obvious fail on GCC/Clang/ICC, so they'll just pile up noise as they error out. Hence why I put them behind a conditional clause.